### PR TITLE
Show validation message on wrong form’s input

### PR DESCRIPTION
**Pull request by jinety**

### CODE REVIEW

File: big-practice/src/js/script.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  }
+};
+
+loginBtn.addEventListener('click', () => {
+  const email = emailInput.value;
+  const password = passwordInput.value;
+
+  if (isEmpty(email) || isEmpty(password)) {
+    warnMsg.innerHTML = MESSAGES.empty;
+  } else if (email === 'admin' && password === 'admin123') {
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/script.js]: Add this account to db.json

File: big-practice/src/js/script.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+const MESSAGES = {
+  empty: 'Email or password is blank. Please enter in full',
+  wrongFormat: 'Wrong email or password. Please re-enter',
+};
+
+/**
+ *  Checks for an empty value
+ */
+const isEmpty = (value) => {
+  if (!value) {
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/script.js]: How about another cases? Try to use if/else in shorthand way: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator

File: big-practice/src/js/script.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+const emailInput = document.getElementById('email');
+const passwordInput = document.getElementById('password');
+const warnMsg = document.getElementById('warnMsg');
+const loginBtn = document.getElementById('loginBtn');
+const emptyText = '';
+
+// Messages
+const MESSAGES = {
+  empty: 'Email or password is blank. Please enter in full',
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/script.js]: Message.empty??? Could you use another name? Btw, message should be Please enter all email and password.

File: big-practice/src/js/script.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
+const emailInput = document.getElementById('email');
+const passwordInput = document.getElementById('password');
+const warnMsg = document.getElementById('warnMsg');
+const loginBtn = document.getElementById('loginBtn');
+const emptyText = '';
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/script.js]: Should be format as PascalCase. Refer: https://www.theserverside.com/answer/Pascal-case-vs-camel-case-Whats-the-difference

File: big-practice/src/style/pages/style-login.css [hunk 5]
```diff
@@ -56,6 +56,12 @@
   cursor: pointer;
 }
 
+.warn-msg {
+  color: red;
```
Comment (phuocduong-agilityio) [inline big-practice/src/style/pages/style-login.css]: Should we define another color set? Ex: --warning-color-100???

File: big-practice/src/js/script.js (hunk 2 above)
Comment (phuocduong-agilityio) [reply big-practice/src/js/script.js]: This haven't fixed yet

File: big-practice/src/js/script.js [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    method: 'GET',
+  };
+  fetch(`${accountApi}/1`, options)
+    .then((response) => response.json())
+    .then((account) => {
+      loginBtn.addEventListener('click', () => {
+        const email = emailInput.value;
+        const password = passwordInput.value;
+
+        isEmpty(email) || isEmpty(password) ? warnMsg.innerHTML = MESSAGES.emptyFormat
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/script.js]: Too complicated, shorthand just applied for simple cases

File: big-practice/src/js/script.js [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+  }
+
+  return false;
+};
+
+const getDataAccount = () => {
+  const options = {
+    method: 'GET',
+  };
+  fetch(`${accountApi}/1`, options)
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/script.js]: Why `1`???

File: big-practice/src/js/script.js [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+const warnMsg = document.getElementById('warnMsg');
+const loginBtn = document.getElementById('loginBtn');
+const accountApi = 'http://localhost:3000/account';
+const EmailRegex = /^([a-zA-Z0-9_\.\-])+\@(([a-zA-Z0-9\-])+\.)+([a-zA-Z0-9]{2,4})+$/;
+const EmptyText = '';
+
+// Messages
+const MESSAGES = {
+  emptyFormat: 'Please enter all email and password',
+  wrongFormat: 'Email entered in the wrong format. Please re-enter',
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/script.js]: What thing is wrong? Email or username?

File: big-practice/src/js/script.js [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+  wrongEmail: 'Login email is incorrect. Please re-enter',
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/script.js]: How can you know when user is wrong password/email?

File: big-practice/src/js/script.js [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+const passwordInput = document.getElementById('password');
+const warnMsg = document.getElementById('warnMsg');
+const loginBtn = document.getElementById('loginBtn');
+const accountApi = 'http://localhost:3000/account';
+const EmailRegex = /^([a-zA-Z0-9_\.\-])+\@(([a-zA-Z0-9\-])+\.)+([a-zA-Z0-9]{2,4})+$/;
+const EmptyText = '';
+
+// Messages
+const MESSAGES = {
+  emptyFormat: 'Please enter all email and password',
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/script.js]: What thing is empty?Btw, empty is not format

File: big-practice/src/js/script.js [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+  }
+};
+
+/**
+ * Email check function is not valid
+ *
+ * @param {string} value - Comparative value
+ */
+const isInvalidEmail = (value) => {
+  if (!EmailRegex.test(value)) {
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/script.js]: Use shorthand condition here

File: big-practice/src/js/script.js [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+    return true;
+  }
+
+  return false;
+};
+
+const getDataAccount = () => {
+  const options = {
+    method: 'GET',
+  };
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/script.js:42]: Add empty line after constant

File: big-practice/src/style/abstracts/color.css [hunk 13]
```diff
@@ -10,4 +10,5 @@
   --secondary-color-200: #7e7373;
   --secondary-color-300: #786e6e;
   --secondary-color-400: #0c0b0b;
+  --warning-color-100: red;
```
Comment (phuocduong-agilityio) [inline big-practice/src/style/abstracts/color.css]: Should be start with 50 or 900??

File: big-practice/src/js/script.js [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+  !value;
+};
+
+/**
+ * Email check function is not valid
+ *
+ * @param {string} value - Comparative value
+ */
+const isValidEmail = (value) => {
+  !!EmailRegex.test(value);
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/script.js]: 1. Missing return
2. Why don't use EmailRegex.test(value) without !!

File: big-practice/src/js/script.js [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+};
+
+/**
+ * Get admin account from json server
+ */
+const getDataAccount = () => {
+  const urlAdmin = `${accountApi}?email=vhoa1000@gmail.com&password=123456`;
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/script.js]: Why do you set email and password here? It should be getting from login form

Btw, try another variable name

File: big-practice/src/js/script.js [hunk 16]
```diff
@@ -0,0 +1,66 @@
...
+
+  fetch(urlAdmin, options)
+    .then((response) => response.json())
+    .then((account) => {
+      loginBtn.addEventListener('click', () => {
+        const email = emailInput.value;
+        const password = passwordInput.value;
+
+        // Email or password cannot be blank
+        if (!isEmpty(email) || !isEmpty(password)) {
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/script.js]: This should be validate before you call request.

File: big-practice/src/js/script.js [hunk 17]
```diff
@@ -0,0 +1,66 @@
...
+ * @param {string} value - Comparative value
+ */
+const isValidEmail = (value) => {
+  !!EmailRegex.test(value);
+};
+
+/**
+ * Get admin account from json server
+ */
+const getDataAccount = () => {
```
Comment (hieungo-agilityio) [inline big-practice/src/js/script.js:36]: func name should be `login`.

File: big-practice/src/js/script.js [hunk 18]
```diff
@@ -0,0 +1,74 @@
...
+const passwordInput = document.getElementById('password');
+const warnMsg = document.getElementById('warnMsg');
+const loginBtn = document.getElementById('loginBtn');
+const accountApi = 'http://localhost:3000/account';
+const EmailRegex = /^([a-zA-Z0-9_\.\-])+\@(([a-zA-Z0-9\-])+\.)+([a-zA-Z0-9]{2,4})+$/;
+const EmptyText = '';
+
+// Messages
+const MESSAGES = {
+  emailPasswordEmpty: 'Please enter all email and password',
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/script.js]: emailPasswordEmpty -> Should be login form empty or something the same

File: big-practice/src/js/script.js [hunk 19]
```diff
@@ -0,0 +1,74 @@
...
+  emailWrongFormat: 'Email entered in the wrong format. Please re-enter',
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/script.js]: Message should be `Email is invalid format`

File: big-practice/src/js/script.js [hunk 20]
```diff
@@ -0,0 +1,74 @@
...
+  errorMessage: 'Email or password is incorrect. Please re-enter',
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/script.js]: What kind of errorMessage???

File: big-practice/src/js/script.js [hunk 21]
```diff
@@ -0,0 +1,74 @@
...
+  const url = `${accountApi}?email=${email}&password=${password}`;
+  const isValidForm = validateForm(email, password);
+
+  if (!isValidForm) {
+    return;
+  }
+
+  fetch(url, { method: 'GET' })
+    .then((response) => response.json())
+    .then((account) => {
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/script.js]: It should be array. Array should be naming as following. Ex: users, products, etc

File: big-practice/db.json [hunk 22]
```diff
@@ -0,0 +1,10 @@
+{
+  "account": [
```
Comment (phuocduong-agilityio) [inline big-practice/db.json]: Should have s after account

File: big-practice/src/js/script.js [hunk 23]
```diff
@@ -0,0 +1,74 @@
...
+const loginBtn = document.getElementById('loginBtn');
+const accountApi = 'http://localhost:3000/accounts';
+const EmailRegex = /^([a-zA-Z0-9_\.\-])+\@(([a-zA-Z0-9\-])+\.)+([a-zA-Z0-9]{2,4})+$/;
+const EmptyText = '';
+
+// Messages
+const MESSAGES = {
+  loginFormEmpty: 'Please enter all email and password',
+  emailWrongFormat: 'Email is invalid format',
+  incorrectLoginForm: 'Email or password is incorrect. Please re-enter',
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/script.js]: incorrectLoginForm -> incorrectLoginAccount

File: big-practice/src/js/script.js [hunk 24]
```diff
@@ -0,0 +1,74 @@
...
+    // Email is not in the correct format
+    warnMsg.innerHTML = MESSAGES.emailWrongFormat;
+  } else {
+    isValidForm = true;
+  }
+
+  return isValidForm;
+};
+
+const loginAdmin = () => {
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/script.js]: 1. Add comment
2. Name should be `login`.

**Pull request merged**